### PR TITLE
Show power

### PR DESCRIPTION
**Pull request by cristianoramon**

### CODE REVIEW

File: wifiphisher/pywifiphisher.py [hunk 1]
```diff
@@ -104,9 +104,18 @@ def parse_args():
...
 
+    parser.add_argument(
+          "-nD",
+          "--noDmasq",
+          help=("To use dmasq itself for dhcp. " +
+                "Example : -nD"
+                ),
+          action='store_true')
+
+
```
Comment (sophron) [inline wifiphisher/pywifiphisher.py]: This is not relevant to this PR's purpose. Please remove it.

File: wifiphisher/pywifiphisher.py [hunk 2]
```diff
@@ -237,6 +246,18 @@ def targeting_cb(pkt):
...
 
     bssid = pkt[Dot11].addr3
+    rssi  = -100
+    # Show signal power
+    if pkt.type == 0 and pkt.subtype == 8:
+        if pkt.haslayer(Dot11Beacon) or pkt.haslayer(Dot11ProbeResp):
+            try:
+                extra = pkt.notdecoded
+                rssi = -(256 - ord(extra[-4:-3]))
+
```
Comment (sophron) [inline wifiphisher/pywifiphisher.py]: There's a blank line here.

File: wifiphisher/pywifiphisher.py [hunk 3]
```diff
@@ -237,6 +246,18 @@ def targeting_cb(pkt):
...
+            except:
+                rssi = -500
+    ######
```
Comment (sophron) [inline wifiphisher/pywifiphisher.py]: Please remove the "######".

File: wifiphisher/pywifiphisher.py [hunk 4]
```diff
@@ -237,6 +246,18 @@ def targeting_cb(pkt):
...
+    rssi  = -100
+    # Show signal power
+    if pkt.type == 0 and pkt.subtype == 8:
+        if pkt.haslayer(Dot11Beacon) or pkt.haslayer(Dot11ProbeResp):
+            try:
+                extra = pkt.notdecoded
+                rssi = -(256 - ord(extra[-4:-3]))
+
+            except:
+                rssi = -500
```
Comment (sophron) [inline wifiphisher/pywifiphisher.py]: Can you tell us a few more things about this? What is this exception is handling exactly?

File: wifiphisher/pywifiphisher.py [hunk 5]
```diff
@@ -281,8 +302,8 @@ def target_APs():
 
     max_name_size = max(map(lambda ap: len(ap[1]), APs.itervalues()))
 
-    header = ('{0:3}  {1:3}  {2:{width}}   {3:19}  {4:14}  {5:}'
-        .format('num', 'ch', 'ESSID', 'BSSID', 'encr', 'vendor', width=max_name_size + 1))
+    header = ('{0:3}  {1:3}  {2:4}  {3:{width}}   {4:19}  {5:14}  {6:}'
+        .format('num', 'ch','power','ESSID', 'BSSID', 'encr', 'vendor', width=max_name_size + 1))
```
Comment (sophron) [inline wifiphisher/pywifiphisher.py]: I think it's better if we put the "power" column at the end (after the "vendor" column).

File: wifiphisher/pywifiphisher.py [hunk 6]
```diff
@@ -458,8 +481,10 @@ def dhcp_conf(interface):
     return '/tmp/dhcpd.conf'
 
 
-def dhcp(dhcpconf, mon_iface):
-    dhcp = Popen(['dnsmasq', '-C', dhcpconf], stdout=PIPE, stderr=DN)
+def dhcp(dhcpconf, mon_iface,args):
+
+    if not args.noDmasq:
+        dhcp = Popen(['dnsmasq', '-C', dhcpconf], stdout=PIPE, stderr=DN)
```
Comment (sophron) [inline wifiphisher/pywifiphisher.py]: This is not relevant to this PR's purpose. Please remove it.

File: wifiphisher/pywifiphisher.py (hunk 1 above)
Comment (cristianoramon) [inline wifiphisher/pywifiphisher.py]: Commit : Show signal strength (power) - adjustment

File: wifiphisher/pywifiphisher.py (hunk 2 above)
Comment (cristianoramon) [inline wifiphisher/pywifiphisher.py]: Commit : Show signal strength (power) - adjustment

File: wifiphisher/pywifiphisher.py (hunk 4 above)
Comment (cristianoramon) [inline wifiphisher/pywifiphisher.py]: Commit : Show signal strength (power) - adjustment

File: wifiphisher/pywifiphisher.py (hunk 3 above)
Comment (cristianoramon) [inline wifiphisher/pywifiphisher.py]: Commit : Show signal strength (power) - adjustment

File: wifiphisher/pywifiphisher.py (hunk 5 above)
Comment (cristianoramon) [inline wifiphisher/pywifiphisher.py]: Commit : Show signal strength (power) - adjustment

File: wifiphisher/pywifiphisher.py (hunk 6 above)
Comment (cristianoramon) [inline wifiphisher/pywifiphisher.py]: Commit : Show signal strength (power) - adjustment

**Comment by blackHatMonkey** [comment]
With the implementation of #470 this PR no longer applies.

**Pull request closed without merging**